File: MaximumSubarray.py
'''
Given an integer array nums, find the subarray with the largest sum, and return its sum.
'''
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curSum = maxSum = nums[0]
        for num in nums[1:]:
            curSum = max(curSum, 0) + num
            maxSum = max(maxSum, curSum)

        return maxSum

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        long sum=0;
        long maxx=Long.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            if(sum>maxx){
                maxx=sum;
            }
            if(sum<0){
                sum=0;
            }
        }
        return (int) maxx;
    }
}